### PR TITLE
Enable CSI Driver by default in helm

### PR DESCRIPTION
**Pull request by albertogdd**
## Description

Make CSI Driver enabled in the default helm chart

## How can this be tested?

`make deploy/helm`

## Checklist

- [X] Unit tests have been updated/added
- [X] PR is labeled accordingly with a single label
- [X] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: config/helm/chart/default/tests/Common/csi/clusterrole-csi_test.yaml
Comment (0sewa0) [inline config/helm/chart/default/tests/Common/csi/clusterrole-csi_test.yaml:5]: update the comment so its not confusing 😅

Comment (0sewa0) [reply config/helm/chart/default/tests/Common/csi/clusterrole-csi_test.yaml:5]: (check other changes for the same 😉 )

**Review by 0sewa0 — CHANGES_REQUESTED** [review]

File: config/helm/chart/default/tests/Common/csi/clusterrole-csi_test.yaml
Comment (albertogdd) [reply config/helm/chart/default/tests/Common/csi/clusterrole-csi_test.yaml]: Fixed in c33e4906a5785055b85428cbefcc4a1a0474a556

**Review by andriisoldatenko — APPROVED** [review]

File: config/helm/chart/default/values.yaml
Comment (luhi-DT) [inline config/helm/chart/default/values.yaml:93]: this would break our current release workflow, as we attach manifests without CSI Driver (see make manifests) :(

**Review by luhi-DT — CHANGES_REQUESTED** [review]

File: config/helm/chart/default/values.yaml
Comment (albertogdd) [reply config/helm/chart/default/values.yaml:93]: :+1: Then, PR blocked by new user story

**Comment by albertogdd** [comment]
Unblocked after merging this PR https://github.com/Dynatrace/dynatrace-operator/pull/2566

File: config/helm/chart/default/values.yaml
Comment (albertogdd) [reply config/helm/chart/default/values.yaml:93]: @luhi-DT https://github.com/Dynatrace/dynatrace-operator/pull/2501#issuecomment-1887075087

**Review by luhi-DT — APPROVED** [review]

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**
